### PR TITLE
Update templates

### PR DESCRIPTION
**Pull request by jmstfv**
* Replace deprecated `devise_error_messages!` (https://github.com/plataformatec/devise/blob/master/CHANGELOG.md) with shared partial
* Standardize devise_invitable templates to look more like Devise templates

This will probably require version bumping to 1.8.0

### CODE REVIEW

File: test/rails_app/app/views/admins/new.html.erb [hunk 1]
```diff
@@ -1,12 +1,16 @@
 <h2>Send invitation</h2>
 
-<%= form_for resource, :as => resource_name, :url => admin_path do |f| %>
+<%= form_for(resource, as: resource_name, url: admin_path) do |f| %>
   <%= devise_error_messages! %>
```
Comment (scambra) [inline test/rails_app/app/views/admins/new.html.erb]: I think it should be changed to render partial

File: test/rails_app/app/views/free_invitations/new.html.erb [hunk 2]
```diff
@@ -1,12 +1,16 @@
 <h2>Send invitation</h2>
 
-<%= form_for resource, :as => resource_name, :url => free_invitation_path do |f| %>
+<%= form_for(resource, as: resource_name, url: free_invitation_path) do |f| %>
   <%= devise_error_messages! %>
```
Comment (scambra) [inline test/rails_app/app/views/free_invitations/new.html.erb]: I think it should be changed to render partial

File: test/rails_app/app/views/users/invitations/new.html.erb [hunk 3]
```diff
@@ -1,15 +1,21 @@
 <h2>Send an invitation</h2>
 
-<%= form_for resource, :as => resource_name, :url => invitation_path(resource_name), :html => { :method => :post } do |f| %>
+<%= form_for(resource, as: resource_name, url: invitation_path(resource_name), html: { method: :post }) do |f| %>
   <%= devise_error_messages! %>
```
Comment (scambra) [inline test/rails_app/app/views/users/invitations/new.html.erb]: I think it should be changed to render partial

File: test/rails_app/app/views/admins/new.html.erb (hunk 1 above)
Comment (jmstfv) [reply test/rails_app/app/views/admins/new.html.erb]: Done! See latest commit (https://github.com/scambra/devise_invitable/pull/786/commits/568ea18b66e879fe3c8ef72ab04ce6981833f61d)

File: test/rails_app/app/views/free_invitations/new.html.erb (hunk 2 above)
Comment (jmstfv) [reply test/rails_app/app/views/free_invitations/new.html.erb]: Done! See latest commit (https://github.com/scambra/devise_invitable/pull/786/commits/568ea18b66e879fe3c8ef72ab04ce6981833f61d)

File: test/rails_app/app/views/users/invitations/new.html.erb (hunk 3 above)
Comment (jmstfv) [reply test/rails_app/app/views/users/invitations/new.html.erb]: Done! See latest commit (https://github.com/scambra/devise_invitable/pull/786/commits/568ea18b66e879fe3c8ef72ab04ce6981833f61d)

**Comment by jmstfv** [comment]
Yeah, keeping partial in sync with devise is less than ideal, but they will probably remove `devise_error_messages!` in the next major release (4.7.0) so I don't see any other options here :worried: 

Your first solution (upgrading minimum devise version, bumping devise_invitable to 1.8.0) seems more sensible to me. I think fiddling with helper is too much work and is error-prone.

**Comment by scambra** [comment]
Helper could be:

```ruby
def devise_invitable_error_messages
  if Devise::VERSION >= '4.6'
    render "devise/shared/error_messages", resource: resource 
  else
    devise_error_messages!
  end
end
```

Although if Devise releases many 4.x versions it would start failing with 4.10.
Also, it could check respond_to?(:devise_error_messages!) && Devise::VERSION =~ /^4\.6\./

Anyway, releasing 1.8.0 without copying partial and require devise >= 4.6 is good enough to me. If some new change is added to devise invitable, people would have to upgrade devise to latest, but it's not hard, it may be recommended for safety, and I think devise 4.6 supports same rails versions as older 4.x devise versions

**Comment by jmstfv** [comment]
If I understood you correctly, you suggest removing shared partial so that devise_invitable inherits the one that comes with devise 4.6+? That actually was my initial idea, but then I inadvertently broke 15 integration tests, so that's why I decided to include the partial.

**Comment by scambra** [comment]
Was it testing against devise 4.6 when integration tests were broken?

**Comment by jmstfv** [comment]
Nope, it was `devise, '~> 4.0'`. With devise 4.6, it _seems_ fine.

**Comment by jmstfv** [comment]
Take a look at https://github.com/scambra/devise_invitable/pull/785. It contains all changes regarding Devise 4.6 and older versions: merge that first. Then I will delete partial from this branch (build will fail, but it is OK) and then merge this one to the master.

**Comment by scambra** [comment]
I have merged #785, you may want to rebase against master

**Comment by jmstfv** [comment]
I have removed shared partial, prepared a changelog for 2.0.0 (since this release will be backwards-incompatible), and fiddled with Travis-CI config. Take a look, but don't merge it yet: I want to update some dependencies. Will let you know when I'm done.

**Pull request merged**